lib/src/Base/Algo/GaussLegendre.cxx

```diff
@@ -0,0 +1,203 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Implement a tensorized Gauss-Legendre quadrature
+ *
+ *  Copyright 2005-2017 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/GaussLegendre.hxx"
+#include "openturns/Tuples.hxx"
+#include "openturns/Exception.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Lapack.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class GaussLegendre
+ */
+
+CLASSNAMEINIT(GaussLegendre);
+
+static const Factory<GaussLegendre> Factory_GaussLegendre;
+
+/* Constructor without parameters */
+GaussLegendre::GaussLegendre(const UnsignedInteger dimension)
+  : IntegrationAlgorithmImplementation()
+  , discretization_(Indices(dimension, ResourceMap::GetAsUnsignedInteger("GaussLegendre-DefaultMarginalIntegrationPointsNumber")))
```

Comment:
@jschueller I had an hesitation, because in ResourceMap we have:
  * GaussLegendre-DefaultMarginalIntegrationPointNumber
  * SpectralModel-DefaultPointNumber
  * Normal-MaximumNumberOfPoints
  * Normal-MinimumNumberOfPoints
  * Student-MaximumNumberOfPoints
  * Student-MinimumNumberOfPoints
  * NumericalMathEvaluation-DefaultPointNumber
  * Distribution-DefaultPointNumber
  * BoxCoxFactory-DefaultPointNumber
So I agree on the fact that a plural would be better, but in this case there are other keys to change. In another PR?